### PR TITLE
Introduce MockToolchain

### PR DESCRIPTION
**Pull request by AndreyAkinshin**
This new toolchain allows writing simple unit tests for summary tables without a need to actually run benchmarks. The usage of the new API is shown in `RatioColumnTest`.

### CODE REVIEW

**Comment by AndreyAkinshin** [comment]
@adamsitnik what do you think?

File: tests/BenchmarkDotNet.Tests/Mocks/MockHelper.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using Xunit.Abstractions;
+
+namespace BenchmarkDotNet.Tests.Mocks
+{
+    public static class MockHelper
+    {
+        public static Summary Run<T>(ITestOutputHelper output)
```
Comment (adamsitnik) [inline tests/BenchmarkDotNet.Tests/Mocks/MockHelper.cs:12]: We could make this method accept a mandatory `Func<BenchmarkCase, List<Measurement>>` argument, pass it to `MockToolchain` instance and remove the need of having a custom attribute, interface and reflection.

**Review by adamsitnik — COMMENTED** [review]
Overall it LGTM, but I believe that we could simplify it.

File: tests/BenchmarkDotNet.Tests/Mocks/MockHelper.cs (hunk 1 above)
Comment (AndreyAkinshin) [reply tests/BenchmarkDotNet.Tests/Mocks/MockHelper.cs]: @adamsitnik good idea, I like it! I rewrote the implementation, could you please check out the new version?

**Comment by AndreyAkinshin** [comment]
@adamsitnik I just merged #2268 to `mocktoolchain`, thanks for the edits!
Can I merge this PR now?

**Review by adamsitnik — APPROVED** [review]
LGTM, and great idea BTW!

**Pull request merged**